tests/samplers_tests/test_samplers.py

```diff
@@ -225,6 +234,120 @@ def sample() -> float:
     np.testing.assert_almost_equal(round_points, points)
 
 
+@parametrize_relative_sampler
+@pytest.mark.parametrize(
+    "x_distribution",
+    [
+        UniformDistribution(-1.0, 1.0),
+        LogUniformDistribution(1e-7, 1.0),
+        DiscreteUniformDistribution(-10, 10, 0.5),
+        IntUniformDistribution(1, 10),
+        IntLogUniformDistribution(1, 100),
+    ],
+)
+@pytest.mark.parametrize(
+    "y_distribution",
+    [
+        UniformDistribution(-1.0, 1.0),
+        LogUniformDistribution(1e-7, 1.0),
+        DiscreteUniformDistribution(-10, 10, 0.5),
+        IntUniformDistribution(1, 10),
+        IntLogUniformDistribution(1, 100),
+    ],
+)
+def test_sample_relative_numerical(
+    relative_sampler_class: Callable[[], BaseSampler],
+    x_distribution: BaseDistribution,
+    y_distribution: BaseDistribution,
+) -> None:
+
+    search_space = {"x": x_distribution, "y": y_distribution}
+    study = optuna.study.create_study(sampler=optuna.samplers.RandomSampler(seed=1))
+    for _ in range(4):
+        trial = study.ask(search_space)
+        study.tell(trial, sum(trial.params.values()))
+    study.sampler = relative_sampler_class()
+
+    values = study.sampler.sample_relative(study, _create_new_trial(study), search_space)
+    for name, distribution in search_space.items():
+        value = values[name]
+        assert isinstance(
+            distribution,
+            (
+                UniformDistribution,
+                LogUniformDistribution,
+                DiscreteUniformDistribution,
+                IntUniformDistribution,
+                IntLogUniformDistribution,
+            ),
+        )
+        assert value >= distribution.low
+        assert value <= distribution.high
```

Comment:
```suggestion
        assert distribution.low <= value <= distribution.high
```

[nits]
It is really nits but it aligns with `test_sample_relative_mixed`.

https://github.com/optuna/optuna/pull/2897/files#diff-8b85ccf8a3534b2aa878bce8ece8ce930af410ab178ebc6d80031badbabb5052R345

Reply:
Good catch! Thank you!